Main.cs

```diff
@@ -64,24 +64,47 @@ public List<Result> Query(Query query)
             }
 
             OpenWindows.Instance.UpdateOpenWindowsList();
-            _results = SearchController.Instance.GetResult(query.Search); 
+            searchResults = SearchController.Instance.GetResult(query.Search); 
 
-            return _results.Select(x => new Result()
-            {
-                Title = x.Result.Title,
-                IcoPath = IconPath,
-                Score = x.Score,
-                SubTitle = $"{Properties.Resources.wox_plugin_windowwalker_running} : {x.Result.ProcessName}",
-                ContextData = x.Result,
-                Action = c =>
-                {
-                    if (c.SpecialKeyState.CtrlPressed)
-                        x.Result.Close();
-                    else 
-                        x.Result.SwitchToWindow();
-                    return true;
-                },
-            }).ToList();
+            var results =
+                    searchResults
+                    .Where(x => !string.IsNullOrEmpty(x.Result.Title))
+                    .Select(x =>
+                        new Result()
+                        {
+                            Title = x.Result.Title,
+                            IcoPath = IconPath,
+                            Score = x.Score,
+                            SubTitle = $"{Properties.Resources.wox_plugin_windowwalker_running} : {x.Result.ProcessName}",
+                            ContextData = x.Result,
+                            Action = c =>
+                            {
+                                if (c.SpecialKeyState.CtrlPressed)
+                                {
+                                    x.Result.Close();
+                                }
+                                else
+                                {
+                                    x.Result.SwitchToWindow();
+                                }
+
+                                return true;
+                            },
+                        })
+                    .ToList();
+
+            var cachedWindowsList = cachedWindows.ToList();
+
+            for (int i = 0; i < results.Count; i++)
+            {
+                for (int c = 0; c < cachedWindowsList.Count; c++)
+                {
+                    if (cachedWindowsList[c].Value.Title == results[i].Title)
+                        results[i].Title = $"{cachedWindowsList[c].Key} - {results[i].Title}";
+                }
+            }
```

Comment:
Is it possible to use join to implement this? Although I don't believe this will be a large performance issue.

Reply:
dont think there is much of a performance impact

Reply:
yes agree with you

Reply:
I havent tried join, feel free to give it a shot and update it. Heading off for now.

Reply:
Yeah I will take a look if I have time, but no need to worry about that.